homeassistant/helpers/device_registry.py

```diff
@@ -853,32 +872,34 @@ def async_update_device(
             area = ar.async_get(self.hass).async_get_or_create(suggested_area)
             area_id = area.id
 
-        if add_config_entry_id is not UNDEFINED:
-            # primary ones have to be at the start.
-            if device_info_type == "primary":
-                # Move entry to first spot
-                if not config_entries or config_entries[0] != add_config_entry_id:
-                    config_entries = [add_config_entry_id] + [
-                        entry
-                        for entry in config_entries
-                        if entry != add_config_entry_id
-                    ]
-
-            # Not primary, append
-            elif add_config_entry_id not in config_entries:
-                config_entries = [*config_entries, add_config_entry_id]
+        if add_config_entry is not UNDEFINED:
+            if (
+                device_info_type == "primary"
+                and add_config_entry.entry_id != old.primary_config_entry
+                and (
+                    old.primary_config_entry is None
+                    or add_config_entry.domain not in LOW_PRIO_CONFIG_ENTRY_DOMAINS
```

Comment:
Following @balloob proposal:
> If REAL primary config entry comes in, it becomes primary if there is a) no primary or b) a SUB primary. 
> If SUB primary config entry comes in, it becomes primary if there is a) no primary

```suggestion
                    or old.primary_config_entry in LOW_PRIO_CONFIG_ENTRY_DOMAINS
```

The current implementation allows overriding the primary if the new one is not in `LOW_PRIO_CONFIG_ENTRY_DOMAINS`

Reply:
That's another way to do it, but I'm not sure why that's better. To avoid flip-flop?